File: test.spritebuilder/Source/TDEngine/internalObject/TowerInternal.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package internalObject;

import Unit.Tower.Tower;
import Utils.Point2D;
 

/**
 *
 * @author vereshchagin
 */
public interface TowerInternal extends ObjectWithID {
 
      double getBulidProgress();
      Boolean getIsBuild();
      enum TowerType{Tower1 ,Tower2,Tower3}
      TowerType getType();
      Point2D  getPosition();
      Point2D  getAttackPoint();
    
    
           
    
}

File: TowerDefence.spritebuilder/internalObject/TowerInternal.swift
//
//  TowerInternal.swift
//  TowerDefence
//
//  Created by koveres on 02.12.14.
//  Copyright (c) 2014 Apportable. All rights reserved.
//

import Foundation

public enum TowerType : CInt{case Tower1 = 21 ,Tower2 = 22,Tower3 = 23

static let allValues = [Tower1, Tower2, Tower3]

}
public protocol TowerInternal : ObjectWithID {
    
    func  getBulidProgress() -> (CDouble);
    func  getIsBuild() -> (Bool);
    func  getType() -> (TowerType);
    func    getPosition() -> (CGPoint);
    func  getAttackPoint() -> (CGPoint?);
    
}
